lib/chai/core/assertions.js

```diff
@@ -988,9 +988,33 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+
+    var target = flag(this, 'object')
+    var validInstanceOfTarget = constructor === Object(constructor) && (
+        typeof constructor === 'function' ||
+        (typeof Symbol !== 'undefined' && typeof Symbol.hasInstance !== 'undefined' &&
+         typeof constructor[Symbol.hasInstance] === 'function')
```

Comment:
Sounds good to me. See https://github.com/chaijs/chai/pull/899#discussion_r95682203 for ideas on only triggering `@@hasInstance` once.